CSharpSourceCode/Battle/AI/Components/WizardAiComponent.cs

```diff
@@ -0,0 +1,70 @@
+﻿using TaleWorlds.Core;
+using TaleWorlds.Library;
+using TaleWorlds.MountAndBlade;
+using TOW_Core.Battle.Extensions;
+using TOW_Core.Utilities;
+
+namespace TOW_Core.Battle.AI.Components
+{
+    public class WizardAiComponent : HumanAIComponent
```

Comment:
is it necessary to extend HumanAiComponent as opposed to just AgentComponent? You are not really using many features of the superclass. That way you wouldnt need to remove it in CustomAiMissionLogic you could just add your new one so they can co-exist on an agent.

Reply:
The problem is, that the Engine code accesses the HumanAIComponent when updating behaviors.
If I dont remove the old one, I would have two components competing to update the behavior of the agent.

If I extend and replace the existing HumanAIComponent, I get the original functionality in its entirety from the baseclass and dont get the "race" between two behavior altering components.

I can explain a bit better once I have finished compiling my documentation on the AI system.